great_expectations/rule_based_profiler/types/domain.py

```diff
@@ -121,11 +121,23 @@ def id(self) -> str:
         return IDDict(self.to_json_dict()).to_id()
 
     def to_json_dict(self) -> dict:
+        details: dict = {}
+
+        key: str
+        semantic_type: Union[str, SemanticDomainTypes]
+        for key, semantic_type in self["details"].items():
+            if isinstance(semantic_type, str):
```

Comment:
Does this work if the string from details is for something other than a SemanticDomainType? Would this code then try to convert it to a SemanticDomainType?

Reply:
It would be great to have a test specifically for this method with and without a semantic_type to be sure.

Reply:
@anthonyburdi It does work -- either `str` or `SemanticType`.  One of my new tests were failing, so this fix (which was a long-standing typo) made them work.  I will add a test (in the next commit).  Thanks.

Reply:
@anthonyburdi I added one test -- I will add another.